### DISCUSSION TITLE
center align text of `taskwarrior`

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [GothenburgBitFactory/taskwarrior#3438] [Q&A]
I found this code but it removes the color and trims if a single task is multiline
```
center() {
  local width
  width="${COLUMNS:-$(tput cols)}"
  while IFS= read -r; do
    # If, by luck, REPLY is the same as width, then just dump it
    (( ${#REPLY} == width )) && printf -- '%s\n' "${REPLY}" && continue

    # Handle lines of any length longer than width
    # this ensures that wrapped overflow is centered
    if (( ${#REPLY} > width )); then
      while read -r subreply; do
        (( ${#subreply} == width )) && printf -- '%s\n' "${subreply}" && continue
        printf -- '%*s\n' $(( (${#subreply} + width) / 2 )) "${subreply}"
      done < <(fold -w "${width}" <<< "${REPLY}")
      continue
    fi

    # Otherwise, print centered
    printf -- '%*s\n' $(( (${#REPLY} + width) / 2 )) "${REPLY}"
  done < "${1:-/dev/stdin}"
}
```

Are there other ways to output tasks? To make it more pretty? like neofetch?

**Comment by wrashi** [comment]
@KAGEYAM4, could you provide more detail about what you're trying to accomplish? Are you running this code in the context  of a larger program or a hook?

**Reply by KAGEYAM4** [reply]
in .bashrc, at the end i execute `task`, if terminal window is fullscreened, the output is left aligned. I wanted the output to be center.

**Reply by wrashi** [reply]
Thanks, that's helpful!

Sadly, my Bash is beyond rusty, but if I'm reading your code right, you're just centering each line. You're not chopping up the columns and centering them. So, when you say, "Make it pretty," what exactly do you have in mind?

**Reply by KAGEYAM4** [reply]
Shouldn't have used the term - 'pretty', i don't know why i used that term.


Just centering should be good enough for me.
